### PR TITLE
Bugfix FXIOS-6245 [v114] fix leading and trailing constraints of Customize Homepage button

### PR DESCRIPTION
**Pull request by AhmadKhattab**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6245)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/14084)

### Description
Changing left and right anchors to leading and trailing to make UI consistent with right-to-left languages and also changing the buttonTrailingSpace value to 18 to make leading and trailing constraints have the same value for a better experience

### Visualizing Code Changes: Screen Shot Provide Clarity on Modifications Made 
after-fix

### Pull requests checks where applicable
- [ ] Fill in the three TODOs above (tickets number and description of your work)
- [ ] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Home/CustomizeHome/CustomizeHomepageSectionCell.swift
Comment (thatswinnie) [inline Client/Frontend/Home/CustomizeHome/CustomizeHomepageSectionCell.swift:13]: The `buttonTrailingSpace` should not be changed. Could you please undo this?

**Review by thatswinnie — COMMENTED** [review]
@AhmadKhattab thank you so much for filing and fixing this. Great work! Could you please make the tiny change I commented on? Other than that it's ready to be merged.

File: Client/Frontend/Home/CustomizeHome/CustomizeHomepageSectionCell.swift
Comment (AhmadKhattab) [reply Client/Frontend/Home/CustomizeHome/CustomizeHomepageSectionCell.swift:13]: Okay, I will roll this change back

Comment (AhmadKhattab) [reply Client/Frontend/Home/CustomizeHome/CustomizeHomepageSectionCell.swift]: Done ✅ Could you please review it again 🙏

**Review by yoanarios — APPROVED** [review]

File: Client/Frontend/Home/CustomizeHome/CustomizeHomepageSectionCell.swift
Comment (yoanarios) [reply Client/Frontend/Home/CustomizeHome/CustomizeHomepageSectionCell.swift]: LGTM thanks for the contribution, I will just wait for @thatswinnie approval to merge

Comment (thatswinnie) [reply Client/Frontend/Home/CustomizeHome/CustomizeHomepageSectionCell.swift]: @AhmadKhattab thanks for the quick change. Unfortunately our build system detected a SwiftLint error on line 52 (the argument is not aligned correctly). Could you please fix this? 🙏

Comment (AhmadKhattab) [reply Client/Frontend/Home/CustomizeHome/CustomizeHomepageSectionCell.swift]: Sure, It's fixed now

**Review by thatswinnie — APPROVED** [review]

**Pull request merged**
